### PR TITLE
Dead references fix

### PR DESCRIPTION
**Pull request by joaopleal**
yo. im tha best

### CODE REVIEW

File: snake.lua [hunk 1]
```diff
@@ -48,6 +48,7 @@ snake.move = function (this,dt)
                 k:onEat(this)
             elseif k.type == "snake" and k ~= this then
                 this:die()
+                return
```
Comment (ingles98) [inline snake.lua:51]: Forgot about that, very good.

File: snake.lua [hunk 2]
```diff
@@ -113,6 +114,14 @@ snake.die = function(this)
         table.removeFromTable(GRID[v.x][v.y], this)
     end
 
+    for x,_v in ipairs(GRID) do
```
Comment (ingles98) [inline snake.lua:117]: This isn't very efficient and doesn't tackle the root issue of not deleting references to the dead snake in the grid, but it does solve the problem

**Review by ingles98 — APPROVED** [review]
Good job m8

File: snake.lua (hunk 1 above)
Comment (joaopleal) [reply snake.lua:51]: Alrightly, glad I could help.

**Pull request merged**